packages/amplify-graphql-sql-transformer/src/graphql-sql-transformer.ts

```diff
@@ -0,0 +1,164 @@
+  MappingTemplate,
+  TransformerPluginBase,
+} from '@aws-amplify/graphql-transformer-core';
+import { TransformerContextProvider, TransformerSchemaVisitStepContextProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import * as cdk from 'aws-cdk-lib';
+import { obj, str, ref, printBlock, compoundExpression, Expression, set, methodCall, ifElse, toJson } from 'graphql-mapping-template';
+import { ResolverResourceIDs, ResourceConstants } from 'graphql-transformer-common';
+import { DirectiveNode, ObjectTypeDefinitionNode, InterfaceTypeDefinitionNode, FieldDefinitionNode } from 'graphql';
+
+type SqlDirectiveConfiguration = {
+  statement: string | undefined;
+  reference: string | undefined;
+  resolverTypeName: string;
+  resolverFieldName: string;
+};
+
+const SQL_DIRECTIVE_STACK = 'CustomSQLStack';
+const directiveDefinition = /* GraphQL */ `
+  directive @sql(statement: String, reference: String) on FIELD_DEFINITION
+`;
+
+export class SqlTransformer extends TransformerPluginBase {
+  private sqlDirectiveFields: Map<FieldDefinitionNode, SqlDirectiveConfiguration[]> = new Map();
+
+  constructor() {
+    super('amplify-sql-transformer', directiveDefinition);
+  }
+
+  field = (
+    parent: ObjectTypeDefinitionNode | InterfaceTypeDefinitionNode,
+    definition: FieldDefinitionNode,
+    directive: DirectiveNode,
+    ctx: TransformerSchemaVisitStepContextProvider,
+  ): void => {
+    if (parent.name.value !== 'Query' && parent.name.value !== 'Mutation') {
+      throw new InvalidDirectiveError(
+        `@sql directive can only be used on Query or Mutation types. Check type "${parent.name.value}" and field "${definition.name.value}".`,
+      );
+    }
+
+    const directiveWrapped = new DirectiveWrapper(directive);
+    if (
+      !directive?.arguments?.find((arg) => arg.name.value === 'statement') &&
+      !directive?.arguments?.find((arg) => arg.name.value === 'reference')
+    ) {
+      throw new InvalidDirectiveError(
+        `@sql directive must have either a 'statement' or 'reference' argument. Check type "${parent.name.value}" and field "${definition.name.value}".`,
+      );
+    }
+
+    const args = directiveWrapped.getArguments(
+      {
+        resolverTypeName: parent.name.value,
+        resolverFieldName: definition.name.value,
+      } as SqlDirectiveConfiguration,
+      generateGetArgumentsInput(ctx.transformParameters),
+    );
+    let resolver = this.sqlDirectiveFields.get(definition);
+
+    if (resolver === undefined) {
+      resolver = [];
+      this.sqlDirectiveFields.set(definition, resolver);
+    }
+
+    resolver.push(args);
+  };
+
+  generateResolvers = (context: TransformerContextProvider): void => {
+    if (this.sqlDirectiveFields.size === 0) {
+      return;
+    }
+
+    const stack: cdk.Stack = context.stackManager.createStack(SQL_DIRECTIVE_STACK);
+    const env = context.synthParameters.amplifyEnvironmentName;
+
+    stack.templateOptions.templateFormatVersion = '2010-09-09';
```

Comment:
Why is this required?

Reply:
This is to handle an edge case where customer hasn't used any `@model` types but using the new SQL directive alone. This makes sure that the stack is created.